src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -206,6 +206,186 @@ bool AletheProofPostprocessCallback::updateTheoryRewriteProofRewriteRule(
                            {},
                            *cdp);
     }
+    // ======== QUANT_VAR_ELIM_EQ
+    //
+    // The conclusion of this rule can take any of the following three forms:
+    //  a) (forall ((x)) (or (not (= x t)) F)) = F{x->t}
+    //  b) (forall ((x)) (or (not (= x t)) F1 ... Fn)) = (or F1 ... Fn){x->t}
+    //  c) (forall ((x)) (not (= x t))) = false
+    //
+    //
+    // In case a:
+    //   Let G = F{x->t},
+    //   r1 = (or (not (= t t)) G)
+    //   rwr = "or-not-refl-empty", t, G
+    //
+    // In case b:
+    //   Let Gi = Fi{x->t},
+    //   F = (or F1 ... Fn),
+    //   G = (or G1 ... Gn),
+    //   r1 = (or (not (= t t)) G1 ... Gn)
+    //   rwr = "or-not-refl"
+    //
+    // Then, for both a and b:
+    //
+    // --------- refl
+    //   VP1_1
```

Comment:
IIUC this would not be a `refl` step, right? In case (a), you'd have that VP1_1 is `(cl (= (or (not (= x t)) F) F{x->t}))`, or am I misunderstanding this? It seems to me that the proof you are building should be under the substitution (i.e., you prove that, in case (a) for example, `x != t v F = F{x->t}` in a context where you have `x->t`).